src/ray/core_worker/store_provider/memory_store/memory_store.cc

```diff
@@ -310,9 +317,35 @@ Status CoreWorkerMemoryStore::GetImpl(const std::vector<ObjectID> &object_ids,
         count += 1;
       } else {
         remaining_ids.insert(object_id);
+        // If reference counting is turned off we can't rely on it to
+        // look up ownership, so we mark this true and take no extra action
+        // based on it.
+        auto has_owner = ref_counter_ == nullptr || ObjectHasOwner(object_id);
```

Comment:
Instead of checking in `CoreWorkerMemoryStore::GetImpl`, can we do it earlier in `CoreWorker::Get`. It's better to do input validation earlier and you have access to `CoreWorker::GetOwnerAddress`.

Reply:
ah this is a good point and helps clean up the code. done. LMK what you think about it now.